adum7701_fmc/clkgen_core.c

```diff
@@ -404,14 +404,10 @@ uint32_t clkgen_get_rate(clkgen_device *dev,
 int32_t clkgen_setup(clkgen_device **clkgen_dev,
 		clkgen_init_params init_params)
 {
-	clkgen_device	*adc_clkgen;
+	clkgen_device	adc_clkgen;
 	uint32_t	reg_val;
 	int32_t		status = 0;
 
-	adc_clkgen = (clkgen_device *)malloc(sizeof(*adc_clkgen));
-        if (!adc_clkgen)
-                return -1;
-
 	adc_clkgen->base_addr = init_params.base_addr;

```

Comment:
if you change the type here, then this also changes to `adc_clkgen.base_addr = init_params.base_addr;`
and you need to call it with `clkgen_set_rate(&adc_clkgen,` and `clkgen_read(&adc_clkgen,`